File: code_wars/6_kyu/persistent_bugger/main.py
def persistence(n):
    n = str(n)
    if len(n) == 1:
        return 0
    from functools import reduce
    string = list(n)
    valor = reduce(lambda a, b: a*b, [int(i) for i in string])
    return 1+persistence(valor)


if __name__ == "__main__":
    assert persistence(39) == 3
    assert persistence(4) == 0
    assert persistence(25) == 2
    assert persistence(999) == 4

File: code_wars/6_kyu/persistent_bugger/main.rs
fn persistence(num: u64) -> u64 {
    if num < 10 {
        return 0;
    }
    let valor = num
        .to_string()
        .chars()
        .map(|f| f.to_string().parse::<u64>().unwrap())
        .product::<u64>();
    return 1 + persistence(valor);
}
fn main() {
    assert_eq!(persistence(39), 3);
    assert_eq!(persistence(4), 0);
    assert_eq!(persistence(25), 2);
    assert_eq!(persistence(999), 4);
}
